src/test/IKImage.test.js

```diff
@@ -291,6 +364,219 @@ describe('IKImage', () => {
+      test('should unmount properly when lazyload is not enabled', () => {
+        const ikImage = shallow(
+          <IKImage
+            urlEndpoint={urlEndpoint}
+            path={relativePath}
+          />
+        );
+        // spies
+        const spy = sinon.spy(ikImage.instance(), 'componentWillUnmount');
+        expect(spy.called).toEqual(false);
+
+        // trigger unmount
+        ikImage.unmount();
+
+        // verify spies
+        expect(spy.calledOnce).toEqual(true);
+        spy.restore();
+
+        expect(ikImage.find('img').length).toEqual(0);
+      });
+    });
+
+    describe('LQIP', () => {
+      // spies
+      const observeSpy = sinon.spy();
+      let intersectionObserverSpy;
+
+      let originalImagePrototype;
+      let imageOnload = null;
+
+      const stubImagePrototype = () => {
+        Object.defineProperty(global.Image.prototype, 'onload', {
+          get: function () {
+            return this._onload;
+          },
+          set: function (fn) {
+            imageOnload = fn;
+            this._onload = fn;
+          },
+          configurable: true
+        });
+      };
+
+      beforeEach(() => {
+        IntersectionObserverMock({ observe: observeSpy });
+        intersectionObserverSpy = sinon.spy(window, 'IntersectionObserver');
+
+        // backup the original Image.prototype.src
+        originalImagePrototype = Object.getOwnPropertyDescriptor(global.Image.prototype, 'src');
+
+        stubImagePrototype();
+      });
+
+      afterEach(() => {
+        intersectionObserverSpy.restore();
+        observeSpy.resetHistory();
+
+        // restore the original Image.prototype.src
+        Object.defineProperty(global.Image.prototype, 'src', originalImagePrototype);
+      });
+
+      test('image with lqip should have actual src when element is loaded', () => {
+        const ikImage = shallow(
+          <IKImage
+            urlEndpoint={urlEndpoint}
+            path={relativePath}
+            lqip={{ active: true }}
+          />
+        );
+
+        const initialURL = `${urlEndpoint}/tr:q-20,bl-6/${relativePath}?${global.SDK_VERSION}`
+        expect(ikImage.find('img').prop('src')).toEqual(initialURL);
+
+        imageOnload();
+
+        const expectedURL = `${urlEndpoint}/${relativePath}?${global.SDK_VERSION}`
+        expect(ikImage.find('img').prop('src')).toEqual(expectedURL);
+      });
+
+      test('image with lqip and lazy loading should have actual src when element is intersected and loaded', () => {
+        const ikImage = mount(
+          <IKImage
+            urlEndpoint={urlEndpoint}
+            path={relativePath}
+            loading="lazy"
+            lqip={{ active: true }}
+          />
+        );
+
+        expect(observeSpy.calledOnce).toEqual(true);
+        expect(intersectionObserverSpy.calledOnce).toEqual(true);
+
+        // trigger element intersection callback
+        intersectionObserverSpy.args[0][0]([{ isIntersecting: true }]);
```

Comment:
lqip image is loaded right away. and then on intersection the original image load is triggered. So you will need to change the order

Reply:
Done.